src/Examine/LuceneEngine/Directories/ExamineDirectory.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using Lucene.Net.Index;
+
+namespace Examine.LuceneEngine.Directories
+{
+    public abstract class ExamineDirectory : Lucene.Net.Store.Directory
+    {
+        private Func<IndexWriter, MergePolicy> _mergePolicy;
+        public bool IsReadOnly { get; set; }
+        public virtual MergePolicy GetMergePolicy(IndexWriter writer)
+        {
+            return _mergePolicy?.Invoke(writer);
+        }
+
+        public void SetMergePolicyAction(Func<IndexWriter, MergePolicy> policy)
```

Comment:
Why do we need public "Set" methods, could these just be part of the ctor parameters?